### PR TITLE
Shrink floating binary point types similar to the Haskell version

### PR DESCRIPTION
**Pull request by moodmosaic**
Based on Hedgehog 0.2.2 `Shrink.hs` module's [towardsFloat](
https://github.com/hedgehogqa/haskell-hedgehog/blob/51ebefc972235d07f25a462b6edbfe3ce507ccbc/hedgehog/src/Hedgehog/Internal/Shrink.hs#L38-L62).

### CODE REVIEW

File: src/Hedgehog/Gen.fs [hunk 1]
```diff
@@ -409,7 +409,7 @@ module Gen =
 
     /// Generates a random 64-bit floating point number.
     let double : Gen<double> =
-        create Shrink.double Random.sizedDouble
+        create (Shrink.towardsDouble 0.0) Random.sizedDouble
```
Comment (moodmosaic) [inline src/Hedgehog/Gen.fs]: I know that at some point we need to change `Gen.double` so that it takes a `Range`, similar to the other primitive combinators. I'd prefer to do this on a separate PR because `Random.sizedDouble` must be changed too and `Seed.nextDouble` must be added first. (Same also for `Gen.float`.)

**Comment by moodmosaic** [comment]
`Range` is in line with the Haskell version, and with this PR `Shrink` too. Next is `Random`, `Seed`, and `Gen`, not necessarily in this order 😃

File: src/Hedgehog/Gen.fs (hunk 1 above)
Comment (jystic) [inline src/Hedgehog/Gen.fs]: 👍

**Comment by jystic** [comment]
Nice 👍
